sote/libsote/rock_layers.lua

```diff
@@ -0,0 +1,114 @@
+local rock_layers = {}
+
+---@enum rock_types
+local rock_types = {
+    no_type         = 0,
+    acid_plutonics  = 3,
+    sandstone       = 4,
+    siltstone       = 5,
+    mudstone        = 7,
+    limestone       = 10, -- 0x0000000A
+    limestone_reef  = 12, -- 0x0000000C
+    basic_volcanics = 22, -- 0x00000016
+    basic_plutonics = 23, -- 0x00000017
+    mixed_volcanics = 24, -- 0x00000018
+    mixed_plutonics = 25, -- 0x00000019
+    acid_volcanics  = 26, -- 0x0000001A
+    slate           = 27, -- 0x0000001B
+    marble          = 28, -- 0x0000001C
+}
+
+for _, value in pairs(rock_types) do
+    rock_layers[value] = {}
+end
+
+require("libsote.stone_layers_loader").load()
+local stone_layers = require("libsote.stone_layer").get_layers_by_id()
+
+local function assign_igneous_extrusive(stone_layer)
+    if stone_layer.acidity < 0.33000001311302185 then
```

Comment:
Can we fix these constants to be more readable? Its a rounding error from original 0.33, wont matter in practice unless you need perfect reproducibility for debugging purposes